KomaMRIBase/src/datatypes/Phantom.jl

```diff
@@ -612,5 +535,59 @@ function check_phantom_arguments(nd, ss, us)
             ssy = ss[2]
         end
     end
+    
     return ssx, ssy, ssz, usx, usy, usz
 end
+
+"""
+    ρ, T1, T2, T2s, Δw = default_brain_tissue_properties(labels, tissue_properties = nothing)
+
+This function returns the default brain tissue properties using a labels identifier Matrix
+# Arguments
+- `labels` : (`::Matrix`) the labels identifier matrix of the phantom
+- `tissue_properties` : (`::Dict`, `=Dict()`) phantom tissue properties in ms and Hz considering the available tissues
+
+# Returns
+- `ρ, T1, T2, T2s, Δw`: (`::Matrix`) matrices of the same size of labels with the tissues properties information
+
+# Examples
+```julia-repl
+julia> ρ, T1, T2, T2s, Δw = default_brain_tissue_properties(labels, tissue_properties)
+
+julia> ρ, T1, T2, T2s, Δw = default_brain_tissue_properties(labels)
+```
+"""
+function default_brain_tissue_properties(labels, tissue_properties = Dict())
+    # Load default tissue properties
+    default_properties = Dict(
+        # ρ, T1, T2, T2*, Δw
+        "CSF"           => [1,      2.569,  0.329,  0.058,  0],
+        "GM"            => [0.86,   0.833,  0.083,  0.069,  0],
+        "WM"            => [0.77,   0.500,  0.070,  0.061,  0],
+        "FAT1"          => [1,      0.350,  0.070,  0.058,  -220*2π], #-220 Hz
+        "MUSCLE"        => [1,      0.900,  0.047,  0.030,  0],
+        "SKIN/MUSCLE"   => [1,      0.569,  0.329,  0.058,  0],
+        "SKULL"         => [0,      0,      0,      0,      0],
+        "VESSELS"       => [0,      0,      0,      0,      0],
+        "FAT2"          => [0.77,   0.500,  0.070,  0.061,  -220*2π], #-220 Hz
+        "DURA"          => [1,      2.569,  0.329,  0.058,  0],
+        "MARROW"        => [0.77,   0.500,  0.070,  0.061,  0])
+    
+    tissue_properties = merge(default_properties, tissue_properties)
+    props = ["ρ", "T1", "T2", "T2s", "Δw"]
+    Nproperties = length(props) 
+    tissue_labels = [23, 46, 70, 93, 116, 139, 162, 185, 209, 232, 255]
+    tissue_texts = ["CSF", "GM", "WM", "FAT1", "MUSCLE", "SKIN/MUSCLE", "SKULL", "VESSELS", "FAT2", "DURA", "MARROW"]
+    data_properties = zeros(Nproperties, size(labels)...)
+    dims = size(data_properties)
+    for i=1:Nproperties
+        for (label, tissue) in zip(tissue_labels, tissue_texts)
+            data_properties[i, ntuple(j -> 1:dims[j+1], length(dims)-1)...] += (labels .== label)*tissue_properties[tissue][i]
+        end
+    end
+    # To make it a tuple of five elements
+    properties = Tuple(data_properties[i, :, :, :] for i in 1:size(data_properties, 1))
+    # To avoid singleton dimensions
+    properties = Tuple(reshape(item, filter(d -> d>1, size(item))...) for item in properties)
+    return properties
```

Comment:
I don't get why this is even necessary `return (data_properties[:, :, :, i] for i in 1:Nproperties)` should be enough. Iterating over `data[i,:,:,:]` instead of `data[:,:,:,i]` is a lot of generating unnecessary code.

Reply:
Again, the problem was the shape inconsistency of the original implementation shape. In this case, `x` and `y` produced 2D matrices, while the `default_brain_tissue_properties` function returned tuples of 3D matrices (with a singleton dimension). It might seem unnecessary, but it was necessary to not modify other parts of the code. Because of that, it made sense to remove the singleton dimension using `Tuple(reshape(item, filter(d -> d>1, size(item))...) for item in properties)`. 

Nevertheless, to don't use those lines, I modified the code.